src/vm/eventpipe.cpp

```diff
@@ -615,7 +615,7 @@ void WINAPI EventPipe::FlushTimer(PVOID parameter, BOOLEAN timerFired)
         // the current timestamp are written into the file.
         LARGE_INTEGER stopTimeStamp;
         QueryPerformanceCounter(&stopTimeStamp);
-        s_pBufferManager->WriteAllBuffersToFile(s_pFile, stopTimeStamp);
+        s_pBufferManager->WriteNextEventToFile(s_pFile, stopTimeStamp);
```

Comment:
Have you tested this on a scenario that produces many events? I think this change limited the IPC channel to sending at most 20 events/sec : ) I recommend that you send all currently buffered events, including partially filled buffers.